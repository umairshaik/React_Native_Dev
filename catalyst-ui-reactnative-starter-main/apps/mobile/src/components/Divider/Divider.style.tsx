import {View} from 'react-native';
import styled from 'styled-components/native';

export const Line = styled(View)`
  height: 0;
  border-bottom-color: #000;
  border-bottom-width: 0.5px;
`;
