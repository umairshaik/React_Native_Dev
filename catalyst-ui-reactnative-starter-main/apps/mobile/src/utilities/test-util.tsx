import {FC, ReactElement, ReactNode} from 'react';
import {render, RenderAPI} from '@testing-library/react-native';
import {QueryClient, QueryClientProvider} from 'react-query';
import {ThemeProvider} from '../context/theme/themeContext';

const query = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      cacheTime: 0,
    },
  },
});
afterEach(() => {
  query.clear();
});
const AllTheProviders: FC<{children: ReactNode}> = ({children}) => (
  <ThemeProvider>
    <QueryClientProvider client={query}>{children}</QueryClientProvider>
  </ThemeProvider>
);

const customRender = (ui: ReactElement, options?: Omit<RenderAPI, 'wrapper'>) =>
  render(ui, {wrapper: AllTheProviders, ...options});

export * from '@testing-library/react-native';
export {customRender as render};
