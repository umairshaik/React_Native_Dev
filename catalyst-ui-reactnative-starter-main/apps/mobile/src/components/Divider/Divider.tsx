import {ViewProps} from 'react-native';
import {Line} from './Divider.style';

type AdditionProps = {
  accessibilityProps: string;
};
type Props = ViewProps & AdditionProps;

const Divider = ({style, accessibilityProps}: Props) => (
  <Line
    style={style}
    accessibilityHint={accessibilityProps}
    accessibilityLabel={`click to ${accessibilityProps}`}
  />
);
export default Divider;
