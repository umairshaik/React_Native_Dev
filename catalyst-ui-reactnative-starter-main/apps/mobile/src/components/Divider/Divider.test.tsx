import Divider from './Divider';
import {render} from '../../utilities/test-util';

describe('[component]-[Divider]', () => {
  test('should render the Divider component', () => {
    const Line = render(<Divider accessibilityProps="mock-divider" />);
    expect(Line).toMatchSnapshot();
  });
});
