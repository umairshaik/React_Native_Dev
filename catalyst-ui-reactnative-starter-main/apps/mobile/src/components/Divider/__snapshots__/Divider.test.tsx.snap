// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component]-[Divider] should render the Divider component 1`] = `
<View
  accessibilityHint="mock-divider"
  accessibilityLabel="click to mock-divider"
  style={
    [
      {
        "borderBottomColor": "#000",
        "borderBottomWidth": 0.5,
        "height": 0,
      },
    ]
  }
/>
`;
